### PR TITLE
Fix ci failure that occurs when testing corner case of ivfpq + refine

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

File: python/run_test.sh [hunk 1]
```diff
@@ -23,15 +23,15 @@ then
...
-python -m spark_rapids_ml tests_no_import_change/test_no_import_change.py 0.2
-# runs on cpu
-python tests_no_import_change/test_no_import_change.py 0.2
-# runs on gpu with spark-submit
-spark-rapids-submit --master local[1] tests_no_import_change/test_no_import_change.py 0.2
-# runs on cpu with spark-submit
-spark-submit --master local[1] tests_no_import_change/test_no_import_change.py 0.2
+## no package import change tests
+## runs on gpu
+#python -m spark_rapids_ml tests_no_import_change/test_no_import_change.py 0.2
```
Comment (eordentlich) [inline python/run_test.sh:28]: Don't forget to undo temporary changes.

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

File: python/tests/test_approximate_nearest_neighbors.py [hunk 2]
```diff
@@ -1046,36 +1046,94 @@ def test_return_fewer_k(
...
         ) -> None:
             assert i == 0 or i == 2
             j = i + 1
-            assert knn_df_collect[i]["indices"] == knn_df_collect[j]["indices"]
-            assert knn_df_collect[i]["distances"] == knn_df_collect[j]["distances"]
-            if knn_df_collect[i]["indices"] == indices_none_probed:
-                assert knn_df_collect[i]["distances"] == distances_none_probed
+            assert res_indices[i] == res_indices[j]
+            assert res_distances[i] == res_distances[j]
+            if res_indices[i] == indices_none_probed:
```
Comment (eordentlich) [inline python/tests/test_approximate_nearest_neighbors.py:1065]: res_indices[i] is an int according to typing and is being compared to list.   Perhaps type is off?

File: python/tests/test_approximate_nearest_neighbors.py [hunk 3]
```diff
@@ -1046,36 +1046,94 @@ def test_return_fewer_k(
...
+        # note that cuVS produces non-deterministic results during the indexing process
         indices_none_probed = [int64_max, int64_max, int64_max, int64_max]
         distances_none_probed = [float_inf, float_inf, float_inf, float_inf]
 
         def check_row_results(
-            i: int, indices_if_probed: List[int], distances_if_probed: List[float]
+            i: int,
+            indices_if_probed: List[int],
+            distances_if_probed: List[float],
+            res_indices: List[int],
```
Comment (eordentlich) [inline python/tests/test_approximate_nearest_neighbors.py:1058]: Looks like these types might be incorrect.   See below. List[List[int]]?

**Comment by lijinf2** [comment]
build

File: python/tests/test_approximate_nearest_neighbors.py (hunk 3 above)
Comment (lijinf2) [reply python/tests/test_approximate_nearest_neighbors.py]: Yeah, revised. Good catch!

File: python/tests/test_approximate_nearest_neighbors.py (hunk 2 above)
Comment (lijinf2) [reply python/tests/test_approximate_nearest_neighbors.py:1065]: This should be a typo and it seems the type checking failed to catch it.

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**